app/app.tsx

```diff
@@ -39,7 +39,7 @@ export const NAVIGATION_PERSISTENCE_KEY = "NAVIGATION_STATE"
  */
 function App() {
   const [ready, setReady] = useState(false)
-  const navigationRef = useRef<NavigationContainerRef>(null)
+  const navigationRef = useRef<NavigationContainerRef<Record<string, never>>>(null)
```

Comment:
TS now required me to set the generic parameter. This parameter refers to something called "parameter" and I think we don't use any yet. It also has to be extended from a JS object, so empty object it is.
`Record<string, never>` is morally equivalent to saying "this is a record with no fields in it" since you can't use/set any values. The type `{}` unfortunately doesn't mean an empty record according to the ESLint error I got:

<img width="1198" alt="Screenshot 2021-08-23 at 20 34 26" src="https://user-images.githubusercontent.com/96054/130499357-cbcdea9f-15cd-4678-8c6c-d0c3c4a92f09.png">

Reply:
I believe that we use parameters, for example, https://github.com/simonwacker/yoga-insights/blob/566b4b019c73ab03170839e1469fcedf1c8c35a4/app/navigators/main-navigator.tsx#L45-L49 defines the parameters for the route `player`. Reading https://reactnavigation.org/docs/typescript/#annotating-ref-on-navigationcontainer it seems to me that the correct type is [`RootParamList`](https://github.com/simonwacker/yoga-insights/blob/566b4b019c73ab03170839e1469fcedf1c8c35a4/app/navigators/root-navigator.tsx#L20-L22). The definition of `RootParamList` is wrong though. I think that instead of `mainStack: undefined` it should be `mainStack: NavigatorScreenParams<MainParamList>` as explained in https://reactnavigation.org/docs/typescript/#type-checking-screens-and-params-in-nested-navigator and already used in https://github.com/simonwacker/yoga-insights/blob/566b4b019c73ab03170839e1469fcedf1c8c35a4/app/navigators/main-navigator.tsx#L43-L44.

Reply:
Thanks! Fixed it with 95766d3. Had to specify the generic parameter of `NavigationContainer` as well: `<NavigationContainer<RootParamList> {...props} ref={ref}>`.

Took me a while to figure out the corresponding error message:

```
$ tsc --noEmit -p . --pretty
app/navigators/root-navigator.tsx:53:37 - error TS2322: Type 'ForwardedRef<NavigationContainerRef<RootParamList>>' is not assignable to type 'Ref<NavigationContainerRef<{}>> | undefined'.
  Type '(instance: NavigationContainerRef<RootParamList> | null) => void' is not assignable to type 'Ref<NavigationContainerRef<{}>> | undefined'.
    Type '(instance: NavigationContainerRef<RootParamList> | null) => void' is not assignable to type '(instance: NavigationContainerRef<{}> | null) => void'.
      Types of parameters 'instance' and 'instance' are incompatible.
        Type 'NavigationContainerRef<{}> | null' is not assignable to type 'NavigationContainerRef<RootParamList> | null'.
          Type 'NavigationContainerRef<{}>' is not assignable to type 'NavigationContainerRef<RootParamList> | null'.
            Type 'NavigationContainerRef<{}>' is not assignable to type 'NavigationContainerRef<RootParamList>'.
              Type 'NavigationContainerRef<{}>' is not assignable to type 'PrivateValueStore<RootParamList, "mainStack", {}>'.
                Types of property '''' are incompatible.
                  Type '{ a: {}; b: never; c: {}; } | undefined' is not assignable to type '{ a: RootParamList; b: "mainStack"; c: {}; } | undefined'.
                    Type '{ a: {}; b: never; c: {}; }' is not assignable to type '{ a: RootParamList; b: "mainStack"; c: {}; }'.
                      Types of property 'a' are incompatible.
                        Property 'mainStack' is missing in type '{}' but required in type 'RootParamList'.

53     <NavigationContainer {...props} ref={ref}>
                                       ~~~

  app/navigators/root-navigator.tsx:25:3
    25   mainStack: NavigatorScreenParams<MainParamList>
         ~~~~~~~~~
    'mainStack' is declared here.
  node_modules/@react-navigation/native/lib/typescript/src/NavigationContainer.d.ts:23:5
    23     ref?: React.Ref<NavigationContainerRef<RootParamList>> | undefined;
           ~~~
    The expected type comes from property 'ref' which is declared here on type 'IntrinsicAttributes & NavigationContainerProps & { theme?: Theme | undefined; linking?: LinkingOptions<{}> | undefined; fallback?: ReactNode; documentTitle?: DocumentTitleOptions | undefined; onReady?: (() => void) | undefined; } & { ...; }'
```

🙈